File: lc_soln/0242_valid_anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        std::vector<int> counter_s(26, 0);
        std::vector<int> counter_t(26, 0);
        for (char c : s) {
            counter_s[c - 'a']++;
        }
        for (char c : t) {
            counter_t[c - 'a']++;
        }
        for (int i = 0; i < 26; ++i) {
            if (counter_s[i] != counter_t[i]) {
                return false;
            }
        }
        return true;
    }
};

int main() {
    return 0;
}

File: lc_soln/0242_valid_anagram/solution.rs
use std::collections::HashMap;

struct Solution;

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }
        let mut counter_s = HashMap::new();
        let mut counter_t = HashMap::new();
        for c in s.chars() {
            *counter_s.entry(c).or_insert(0) += 1;
        }
        for c in t.chars() {
            *counter_t.entry(c).or_insert(0) += 1;
        }
        counter_s == counter_t
    }
}

fn main() {}
